lib/phoenix_live_view/test/structs.ex

```diff
@@ -41,5 +41,63 @@ defmodule Phoenix.LiveViewTest.Element do
             selector: nil,
             text_filter: nil,
             event: nil,
-            form_data: nil
+            form_data: nil,
+            meta: %{}
+end
+
+defmodule Phoenix.LiveViewTest.Upload do
+  @moduledoc """
+  The struct returned by `Phoenix.LiveViewTest.file_input/4`.
+
+  The following public fields represent the element:
+
+    * `selector` - The query selector
+    * `entries` - The list of selected file entries
+
+  See the `Phoenix.LiveViewTest` documentation for usage.
+  """
+
+  alias Phoenix.LiveViewTest.{Upload, Element}
+
+  if Version.match?(System.version(), ">= 1.8.0") do
+    @derive {Inspect, only: [:selector, :entries]}
+  end
+
+  defstruct pid: nil,
+            view: nil,
+            element: nil,
+            ref: nil,
+            selector: nil,
+            config: %{},
+            entries: []
+
+  @doc false
+  def new(pid, %Phoenix.LiveViewTest.View{} = view, form_selector, name, entries) do
+    populated_entries = Enum.map(entries, fn entry -> populate_entry(entry) end)
+    selector = "#{form_selector} input[type=\"file\"][name=\"#{name}\"]"
+
+    %Upload{
+      pid: pid,
+      view: view,
+      element: %Element{proxy: view.proxy, selector: selector},
+      entries: populated_entries,
+    }
+  end
+
+  defp populate_entry(%{} = entry) do
+    name =
+      Map.get(entry, :name) ||
+        raise ArgumentError, "a :name of the entry filename is required."
+
+    content =
+      Map.get(entry, :content) ||
+        raise ArgumentError, "the :content of the binary entry file data is required."
+
+    %{}
```

Comment:
Since this is a new map, can't we do:

```
    %{"name" => name, "content" => content, "ref" => System.unique_integer([:positive], "size" => byte_size(content), "type" =>  MIME.from_path(name)}`

```

Reply:
we can half populate the map instead of the `put`'s, but we need the put_new's because the test clients can intentionally send  hard-coded size/mime values to test constraint violations

Reply:
If you use `put_new_lazy` on a new map, it will always evaluate though, right?

```
Map.put_new_lazy(%{}, "size", fn -> 100 end)
%{"size" => 100}
```